From e63424b2566cf0eb18afee851080d2a7e5968412 Mon Sep 17 00:00:00 2001
From: Jonas Hahnfeld <jonas.hahnfeld@cern.ch>
Date: Thu, 19 Jan 2023 14:39:37 +0100
Subject: [PATCH 5/8] [clang] Fix up application of D41416

The LLVM 13 port of the patch missed to adapt findSpecialization of
ClassTemplateDecl and VarTemplateDecl. During the upgrade, we only
noticed and fixed FunctionTemplateDecl::findSpecialization.

(cherry picked from commit 41840bfaccb50898664b6396cc3a907932a1b8f4)
---
 interpreter/llvm/src/tools/clang/lib/AST/DeclTemplate.cpp | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/interpreter/llvm/src/tools/clang/lib/AST/DeclTemplate.cpp b/interpreter/llvm/src/tools/clang/lib/AST/DeclTemplate.cpp
index 8fb204ef7d..c344a16ab4 100644
--- a/interpreter/llvm/src/tools/clang/lib/AST/DeclTemplate.cpp
+++ b/interpreter/llvm/src/tools/clang/lib/AST/DeclTemplate.cpp
@@ -519,7 +519,8 @@ ClassTemplateDecl::newCommon(ASTContext &C) const {
 ClassTemplateSpecializationDecl *
 ClassTemplateDecl::findSpecialization(ArrayRef<TemplateArgument> Args,
                                       void *&InsertPos) {
-  return findSpecializationImpl(getSpecializations(), InsertPos, Args);
+  auto *Common = getCommonPtr();
+  return findSpecializationImpl(Common->Specializations, InsertPos, Args);
 }
 
 void ClassTemplateDecl::AddSpecialization(ClassTemplateSpecializationDecl *D,
@@ -1236,7 +1237,8 @@ VarTemplateDecl::newCommon(ASTContext &C) const {
 VarTemplateSpecializationDecl *
 VarTemplateDecl::findSpecialization(ArrayRef<TemplateArgument> Args,
                                     void *&InsertPos) {
-  return findSpecializationImpl(getSpecializations(), InsertPos, Args);
+  auto *Common = getCommonPtr();
+  return findSpecializationImpl(Common->Specializations, InsertPos, Args);
 }
 
 void VarTemplateDecl::AddSpecialization(VarTemplateSpecializationDecl *D,
-- 
2.42.0

